pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java

```diff
@@ -0,0 +1,408 @@
+package org.broadinstitute.dsm.db;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.dsm.db.dao.ddp.instance.DDPInstanceDao;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.db.structure.ColumnName;
+import org.broadinstitute.dsm.db.structure.TableName;
+import org.broadinstitute.dsm.model.elastic.export.Exportable;
+import org.broadinstitute.dsm.model.elastic.export.painless.PutToNestedScriptBuilder;
+import org.broadinstitute.dsm.model.elastic.export.painless.UpsertPainlessFacade;
+import org.broadinstitute.dsm.model.filter.postfilter.HasDdpInstanceId;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.lddp.db.SimpleResult;
+
+@Slf4j
+@Data
+@TableName(name = DBConstants.SOMATIC_DOCUMENTS_TABLE,
+        alias = DBConstants.SOMATIC_DOCUMENTS_TABLE_ALIAS,
+        primaryKey = DBConstants.SOMATIC_DOCUMENTS_PK,
+        columnPrefix = "")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class SomaticResultUpload implements HasDdpInstanceId {
+    @ColumnName(ESObjectConstants.SOMATIC_DOCUMENT_ID)
+    private final Long somaticDocumentId;
+
+    @ColumnName(DBConstants.DDP_INSTANCE_ID)
+    private final Long ddpInstanceId;
+
+    private final String ddpParticipantId;
+
+    @ColumnName(DBConstants.PARTICIPANT_ID)
+    private final Long participantId;
+
+    @ColumnName(DBConstants.FILE_NAME)
+    private final String fileName;
+
+    @ColumnName(DBConstants.MIME_TYPE)
+    private final String mimeType;
+
+    private final String bucket;
+
+    private final String blobPath;
+
+    @ColumnName(DBConstants.CREATED_BY_USER_ID)
+    private final Long createdByUserId;
+
+    @ColumnName(DBConstants.CREATED_AT)
+    private final long createdAt;
+
+    @ColumnName(DBConstants.DELETED_BY_USER_ID)
+    private final Long deletedByUserId;
+
+    @ColumnName(DBConstants.DELETED_AT)
+    private final long deletedAt;
```

Comment:
It looks like this becomes a legacy audit record once the file is deleted, but the queries below do not account for that?

Reply:
They do not.  We will let the CRCs know that a file has been uploaded and since deleted.

Reply:
Maybe there is nothing to do here, but I imagine for many uses of these queries the caller will have to then filter out deleted files. Is that not the case?

Reply:
The caller will be able to answer the question:
What are the somatic results that have been provided to this participant.  Some may have been deleted (by users or by virus scanning) - and the study staff may be fielding phone calls related to the deleted files since a follow-up survey may have been created.
It's trivial to filter out the deleted files.

Happy to implement a change here.  What would you like to see?

Reply:
If you feel the main use case is to return deleted entries along with live ones, then no change is needed. In any case, if you've thought about the use cases and are comfortable with this implementation, then so am I. Just wanted to be sure you understood my concern.